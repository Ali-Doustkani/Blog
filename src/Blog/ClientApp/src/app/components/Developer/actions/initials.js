const STATUS = {
   LOADING: 1,
   SAVING: 2,
   IDLE: 3,
   STOPPED: 4
}

const initialState = {
   status: STATUS.LOADING,
   errorMessage: '',
   summary: '',
   summaryErrors: [],
   skills: '',
   skillsErrors: [],
   experiences: [],
   experiencesErrors: [],
   sideProjects: [],
   sideProjectErrors: []
}

export { initialState, STATUS }
