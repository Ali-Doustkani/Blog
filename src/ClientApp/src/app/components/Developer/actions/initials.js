const STATUS = {
   LOADING: 1,
   SAVING: 2,
   IDLE: 3,
   STOPPED: 4
}

const initialState = {
   status: STATUS.LOADING,
   errorMessage: '',
   summary: '',
   skills: '',
   experiences: [],
   sideProjects: [],
   educations: []
}

export { initialState, STATUS }
