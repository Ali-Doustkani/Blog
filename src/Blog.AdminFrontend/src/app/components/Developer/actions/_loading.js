import { initialState, STATUS } from './initials'
import { initialExperienceErrors } from './_experiences'
import { evolve, evolveFrom, take, ifElse } from '../../../utils/fn'

const loadOK = ifElse(
   action => action.result.data,
   evolve(
      take(op => op.action.result.data),
      take(op => ({
         experiences: op.action.result.data.experiences.map(initialExperienceErrors)
      })),
      { status: STATUS.IDLE }
   ),
   evolveFrom(initialState, { status: STATUS.IDLE })
)

const loadFatal = evolve(take(op => ({ errorMessage: op.action.result.data })), {
   status: STATUS.STOPPED
})

const load = ifElse(action => action.result.status === 'ok', loadOK, loadFatal)

export { load }
