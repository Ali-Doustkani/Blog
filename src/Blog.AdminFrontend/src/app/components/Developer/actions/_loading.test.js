import { load } from './_loading'
import { initialState } from './initials'

it('create default values when no developer exists', () => {
   const newState = load(initialState, {
      result: {
         status: 'ok',
         data: null
      }
   })

   expect(newState).toMatchSnapshot()
})

it('put form in STOPPED mode when fatal happens', () => {
   const newState = load(initialState, {
      result: {
         status: 'fatal',
         data: 'Error Happened'
      }
   })

   expect(newState).toMatchSnapshot()
})

it('stores experience format', () => {
   const state = load(initialState, {
      result: {
         status: 'ok',
         data: {
            experiences: [{ id: 1, company: 'Lodgify' }]
         }
      }
   })

   expect(state).toMatchSnapshot()
})
