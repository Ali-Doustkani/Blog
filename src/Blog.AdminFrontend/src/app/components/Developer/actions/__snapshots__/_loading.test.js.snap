// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create default values when no developer exists 1`] = `
Object {
  "errorMessage": "",
  "experiences": Array [],
  "experiencesErrors": Array [],
  "skills": "",
  "skillsErrors": Array [],
  "status": 3,
  "summary": "",
  "summaryErrors": Array [],
}
`;

exports[`put form in STOPPED mode when fatal happens 1`] = `
Object {
  "errorMessage": "Error Happened",
  "experiences": Array [],
  "experiencesErrors": Array [],
  "skills": "",
  "skillsErrors": Array [],
  "status": 4,
  "summary": "",
  "summaryErrors": Array [],
}
`;

exports[`stores experience format 1`] = `
Object {
  "errorMessage": "",
  "experiences": Array [
    Object {
      "company": "Lodgify",
      "companyErrors": Array [],
      "contentErrors": Array [],
      "endDateErrors": Array [],
      "id": 1,
      "positionErrors": Array [],
      "startDateErrors": Array [],
    },
  ],
  "experiencesErrors": Array [],
  "skills": "",
  "skillsErrors": Array [],
  "status": 3,
  "summary": "",
  "summaryErrors": Array [],
}
`;
